src/pudl_archiver/depositors/fsspec.py

```diff
@@ -0,0 +1,272 @@
+        )
+
+    async def create_file(
+        self,
+        filename: str,
+        data: BinaryIO,
+    ) -> "FsspecDraftDeposition":
+        """Create a file in a deposition."""
+        with (self.deposition.deposition_path / filename).open(mode="wb") as f:
+            f.write(data.read())
+
+        return self.model_copy(
+            update={
+                "deposition": Deposition.from_upath(self.deposition.deposition_path)
+            }
+        )
+
+    async def delete_file(self, filename: str) -> "FsspecDraftDeposition":
+        """Delete a file from a deposition."""
+        (self.deposition.deposition_path / filename).unlink()
+        return self.model_copy(
+            update={
+                "deposition": Deposition.from_upath(self.deposition.deposition_path),
+            }
+        )
+
+    async def get_file(self, filename: str) -> bytes:
+        """Download file from deposition."""
+        return self.api_client.get_file(self.deposition, filename)
+
+    async def cleanup_after_error(self, e: Exception):
+        """Cleanup draft after an error during an archive run."""
+        logger.error(
+            f"Failed while creating new deposition: {traceback.print_exception(e)}"
+        )
+
+    async def delete_deposition(self) -> None:
+        """Delete an un-submitted deposition."""
+        pass
+
+    def generate_change(
+        self, filename: str, resource: ResourceInfo
+    ) -> DepositionChange:
+        """Check whether file exists and should be changed."""
+        remote_path = self.deposition.deposition_path / filename
+
+        if remote_path.exists():
+            remote_md5 = compute_md5(remote_path)
+            local_md5 = compute_md5(resource.local_path)
+            if remote_md5 != local_md5:
+                logger.info(
+                    f"Updating {filename}: local hash {local_md5} vs. remote {remote_md5}"
+                )
+                action = DepositionAction.UPDATE
+            else:
+                action = DepositionAction.NO_OP
+        else:
+            logger.info(f"Adding {filename} to deposition.")
+
+            action = DepositionAction.CREATE
+
+        return DepositionChange(
+            action_type=action,
+            name=filename,
+            resource=resource.local_path,
+        )
+
+    def generate_datapackage(
+        self, resource_info: dict[str, ResourceInfo]
+    ) -> DataPackage:
+        """Generate new datapackage, attach to deposition, and return."""
+        logger.info(f"Creating new datapackage.json for {self.dataset_id}")
+
+        # Create updated datapackage
+        resources = [
+            _resource_from_upath(
+                self.deposition.deposition_path / fname, resource_info[fname].partitions
+            )
+            for fname in self.deposition.file_list
+            if fname != "datapackage.json"
+        ]
+        datapackage = DataPackage.mecs(
```

Comment:
I've changed this to call a generic `new_datapackage` method and added some slightly janky logic to that method to handle the `mecs` case. In general, it would be good to not have to get metadata from PUDL so we could handle this more cleanly, but that's a bigger rabbit hole.